File: backend/semgrep_rules/tests/javascript/kisa_sw_41_use_after_free.js
function unsafe(native, handle) {
  native.free(handle);
  // ruleid: kisa.sw41.javascript.native-handle-use-after-free-review
  return native.read(handle);
}
function safe(native, handle) {
  const value = native.read(handle);
  native.free(handle);
  // ok: kisa.sw41.javascript.native-handle-use-after-free-review
  return value;
}

File: backend/semgrep_rules/tests/java/kisa_sw_41_use_after_free.java
class kisa_sw_41_use_after_free {
    byte unsafe(sun.misc.Unsafe u, long addr) {
        u.freeMemory(addr);
        // ruleid: kisa.sw41.java.unsafe-use-after-free-review
        return u.getByte(addr);
    }
    byte safe(sun.misc.Unsafe u, long addr) {
        byte value = u.getByte(addr);
        u.freeMemory(addr);
        // ok: kisa.sw41.java.unsafe-use-after-free-review
        return value;
    }
}
